run.go

```diff
@@ -120,13 +152,13 @@ func validateRunOpts(ro runOptions) error {
 	return nil
 }
 
-func signCollection(collection attestation.Collection, opts ...dsse.SignOption) (dsse.Envelope, error) {
+func createAndSignEnvelope(collection interface{}, predType string, subjects map[string]cryptoutil.DigestSet, opts ...dsse.SignOption) (dsse.Envelope, error) {
```

Comment:
I think maybe keeping `signCollection` as is could be the right way here, and then maybe break out the contents to another shared function? 🤷 at the same time this might just be me getting the heebie jeebies by the fact that it still says `collection` for the input var. If this was change to something like `predicate`, maybe it would make more sense?

Reply:
Yeah, renaming to `predicate` makes sense.